seg_utils/ui/dialogs.py

```diff
@@ -99,6 +159,264 @@ def __init__(self, shape: str, *args):
+        font.setKerning(True)
+        self.shapeText.setFont(font)
+        self.shapeText.setPlaceholderText("Enter new Shape name")
+        self.shapeText.setMaximumHeight(25)
+
+        # Buttons
+        self.buttonWidget = QDialogButtonBox(QDialogButtonBox.Ok | QDialogButtonBox.Cancel)
+        self.buttonWidget.accepted.connect(self.ok_clicked)
+        self.buttonWidget.rejected.connect(self.close)
+
+        layout = QVBoxLayout(self)
+        layout.addWidget(self.shapeText)
+        layout.addWidget(self.buttonWidget)
+
+    def ok_clicked(self):
+        """ only accepts the new label class if
+        (1) the name doesn't already exist and
+        (2) the name is not an empty string"""
+        name = self.shapeText.toPlainText()
+        if name in self.existing_classes:
+            self.shapeText.setText("")
+            self.shapeText.setPlaceholderText("'{}' already exists".format(name))
+        elif name:
+            self.new_label_class = name
+            self.close()
+
+
+class SelectFileTypeDialog(QDialog):
+    """
+    handles a QDialog to let the user select between video, image and whole slide image
+    the selected type is the used when importing a new file
+    """
+    def __init__(self):
+        super(SelectFileTypeDialog, self).__init__()
+
+        self.setFixedSize(QSize(250, 150))
+        self.setWindowTitle("Select File Type")
+        self.filetype = ""
+
+        # create buttons and apply stylesheet
+        v = QPushButton("Video")
+        i = QPushButton("Image")
+        w = QPushButton("Whole Slide Image")
+        v.setStyleSheet(BUTTON_STYLESHEET)
+        i.setStyleSheet(BUTTON_STYLESHEET)
+        w.setStyleSheet(BUTTON_STYLESHEET)
+
+        # TODO: Extend by other accepted types, substitute 'whatever' by actual WSI type
+        v.clicked.connect(lambda: self.set_type("mp4"))
+        i.clicked.connect(lambda: self.set_type("png"))
+        w.clicked.connect(lambda: self.set_type("whatever"))
+
+        ft_layout = QVBoxLayout(self)
+        ft_layout.addWidget(v)
+        ft_layout.addWidget(i)
+        ft_layout.addWidget(w)
+
+    def set_type(self, t: str):
+        """ updates the class variable by the selected filetype"""
+        self.filetype = t
+        self.close()
+
+
+class ProjectHandlerDialog(QDialog):
+    def __init__(self, parent):
+        super(ProjectHandlerDialog, self).__init__()
+        self.setFixedSize(parent.size() / 2)
+        self.setWindowTitle('Create new Project')
+
+        self.project_path = ""
+        self.files = list()
+
+        # Header for LineEdit
+        self.header = QLabel()
+        self.header.setStyleSheet("font: bold 12px")
+        self.header.setText("Choose Project Location")
+
+        # LineEdit where user can enter a path
+        self.enter_path = QLineEdit(self)
+        self.enter_path.setFixedHeight(30)
+
+        # suggestion for a project location
+        suggestion = str(Path.home()) + '/AnnotationProjects/project'
```

Comment:
I really like that you do this. It would be cleaner to use an f-string like this:
`suggestion = f"{Path.home()}{os.path.sep}AnnotationProjects{os.path.sep}project"`